recipes/theora/all/conanfile.py

```diff
@@ -1,182 +1,105 @@
-                self.build_requires("msys2/cci.latest")
-
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version][0], strip_root=True, destination=self._source_subfolder)
-
-        source = self.conan_data["sources"][self.version][1]
-        url = source["url"]
-        filename = url[url.rfind("/") + 1:]
-        tools.download(url, filename)
-        tools.check_sha256(filename, source["sha256"])
-
-        shutil.move(filename, os.path.join(self._source_subfolder, "lib", filename))
-
-    def _build_msvc(self):
-        def format_libs(libs):
-            return " ".join([l + ".lib" for l in libs])
-
-        project = "libtheora"
-        config = "dynamic" if self.options.shared else "static"
-        sln_dir = os.path.join(self._source_subfolder, "win32", "VS2008")
-        vcproj_path = os.path.join(sln_dir, project, "{}_{}.vcproj".format(project, config))
-
-        # fix hard-coded ogg names
-        if self.options.shared:
-            tools.replace_in_file(vcproj_path,
-                                  "libogg.lib",
-                                  format_libs(self.deps_cpp_info["ogg"].libs))
-
-        # Honor vc runtime from profile
-        if "MT" in msvc_runtime_flag(self):
-            tools.replace_in_file(vcproj_path, 'RuntimeLibrary="2"', 'RuntimeLibrary="0"')
-            tools.replace_in_file(vcproj_path, 'RuntimeLibrary="3"', 'RuntimeLibrary="1"')
-
-        sln = "{}_{}.sln".format(project, config)
-        targets = ["libtheora" if self.options.shared else "libtheora_static"]
-        properties = {
-            # Enable LTO when CFLAGS contains -GL
-            "WholeProgramOptimization": "true" if any(re.finditer("(^| )[/-]GL($| )", tools.get_env("CFLAGS", ""))) else "false",
-        }
-
-        with tools.chdir(sln_dir):
-            msbuild = MSBuild(self)
-            msbuild.build(sln, targets=targets, platforms={"x86": "Win32", "x86_64": "x64"}, properties=properties)
-
-    @functools.lru_cache(1)
-    def _configure_autotools(self):
-        autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
-        yes_no = lambda v: "yes" if v else "no"
-        args = [
-            "--enable-shared={}".format(yes_no(self.options.shared)),
-            "--enable-static={}".format(yes_no(not self.options.shared)),
-            "--disable-examples",
-        ]
-        autotools.configure(configure_dir=self._source_subfolder, args=args)
-        return autotools
+        get(self, **self.conan_data["sources"][self.version][0], destination=self.source_folder, strip_root=True)
+
+        def_source = self.conan_data["sources"][self.version][1]
+        def_url = def_source["url"]
+        def_filename = def_url[def_url.rfind("/") + 1:]
+        def_path = os.path.join(self.source_folder, "lib", def_filename)
+        download(self, **def_source, filename=def_path)
+        replace_in_file(self, def_path, "LIBRARY	libtheora", "LIBRARY	theora")
```

Comment:
Patching must be done in the build method

Reply:
oops